pkg/roll/execute.go

```diff
@@ -109,16 +109,15 @@ func (m *Roll) StartDDLOperations(ctx context.Context, migration *migrations.Mig
 			continue
 		}
 
-		for _, action := range startOp.Actions {
-			if err := action.Execute(ctx); err != nil {
-				errRollback := m.Rollback(ctx)
-				if errRollback != nil {
-					return nil, errors.Join(
-						fmt.Errorf("unable to execute start operation of %q: %w", migration.Name, err),
-						fmt.Errorf("unable to roll back failed operation: %w", errRollback))
-				}
-				return nil, fmt.Errorf("failed to start %q migration, changes rolled back: %w", migration.Name, err)
+		coordinator := migrations.NewCoordinator(startOp.Actions)
```

Comment:
I would like to move the coordinator one step later, so it collects all actions for all operations in a follow-up PR.

Reply:
what is the issue that prevents doing us doing so at the moment?